kfs-cam/src/main/java/edu/arizona/kfs/module/cam/document/validation/impl/AssetRule.java

```diff
@@ -0,0 +1,75 @@
+package edu.arizona.kfs.module.cam.document.validation.impl;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.kuali.kfs.module.cam.businessobject.Asset;
+import org.kuali.kfs.sys.KFSConstants;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.kns.document.MaintenanceDocument;
+import org.kuali.rice.krad.service.BusinessObjectService;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+import edu.arizona.kfs.module.cam.CamsConstants;
+import edu.arizona.kfs.module.cam.CamsKeyConstants;
+import edu.arizona.kfs.module.cam.CamsPropertyConstants;
+import edu.arizona.kfs.module.cam.businessobject.AssetExtension;
+import edu.arizona.kfs.module.cam.businessobject.AssetInventoryUnit;
+import edu.arizona.kfs.sys.KFSPropertyConstants;
+
+public class AssetRule extends org.kuali.kfs.module.cam.document.validation.impl.AssetRule {
+
+	protected boolean isDocumentValidForSave(MaintenanceDocument document) {
+		boolean valid = super.isDocumentValidForSave(document);
```

Comment:
We should check on the success of our super call before going on with our UA specific validations:
`if (!valid) {
    super.resumeErrorPath();
    return false;
}`